recipes/HeterogenousArrayLog/src/Main.purs

```diff
@@ -0,0 +1,87 @@
+module HeterogenousArrayLog.Main where
+
+import Prelude
+
+import Data.Functor.Variant (SProxy(..))
+import Data.Variant (Variant)
+import Data.Variant as Variant
+import Effect (Effect, foreachE)
+import Effect.Console (log)
+
+separateWithDashes :: Effect Unit
+separateWithDashes = log "------------------"
+
+main :: Effect Unit
+main = do
+  {-
+    PureScript arrays must store values that have the same type. Attempting
+    to store values of different types will result in a compiler error.
+
+    To get around this limitation while still ensuring type safety, we can use
+    `Variant`.
```

Comment:
I think this note along with some more explanation could go in the readme. Content after line three won't be added to the table, so we can feel free to be more verbose in there.
I can see this topic being a bit unapproachable for beginners if the cookbook is their first exposure.

Reply:
Good point.

Do you think we should also indicate difficulty in the recipe?